File: app/src/main/java/com/mapmyindia/sdk/demo/java/adapter/AutoSuggestSearchesAdapter.java
package com.mapmyindia.sdk.demo.java.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.mapmyindia.sdk.demo.R;
import com.mmi.services.api.autosuggest.model.SuggestedSearchAtlas;

import java.util.ArrayList;

/**
 * Created by CEINFO on 26-02-2019.
 */

public class AutoSuggestSearchesAdapter extends RecyclerView.Adapter<AutoSuggestSearchesAdapter.MyViewholder> {
    private ArrayList<SuggestedSearchAtlas> list;
    private SuggestedSearches suggestedSearches;

    public   AutoSuggestSearchesAdapter(ArrayList<SuggestedSearchAtlas> list, SuggestedSearches suggestedSearches) {
        this.list = list;
        this.suggestedSearches = suggestedSearches;
    }

    @NonNull
    @Override
    public MyViewholder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.auto_suggest_adapter_row, parent, false);
        return new MyViewholder(v);

    }

    @Override
    public void onBindViewHolder(@NonNull MyViewholder holder, final int position) {
        holder.viewName.setText(list.get(position).location);

        holder.viewName.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                suggestedSearches.hyperlinkData(list.get(position).hyperLink);
            }
        });
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public interface SuggestedSearches {
        void hyperlinkData(String hyperlink);
    }

    class MyViewholder extends RecyclerView.ViewHolder {

        TextView viewName;

        public MyViewholder(View itemView) {
            super(itemView);
            viewName = itemView.findViewById(R.id.textView);
        }
    }
}

File: app/src/main/java/com/mapmyindia/sdk/demo/kotlin/adapter/AutoSuggestSearchesAdapter.kt
package com.mapmyindia.sdk.demo.kotlin.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.mapmyindia.sdk.demo.R
import com.mmi.services.api.autosuggest.model.SuggestedSearchAtlas
import java.util.*

class AutoSuggestSearchesAdapter(private val list: ArrayList<SuggestedSearchAtlas>, private val suggestedSearches: SuggestedSearches) : RecyclerView.Adapter<AutoSuggestSearchesAdapter.MyViewholder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewholder {
        val v: View = LayoutInflater.from(parent.context).inflate(R.layout.auto_suggest_adapter_row, parent, false)
        return MyViewholder(v)
    }

    override fun onBindViewHolder(holder: MyViewholder, position: Int) {
        holder.viewName.text = list[position].location
        holder.viewName.setOnClickListener { suggestedSearches.hyperlinkData(list[position].hyperLink) }
    }

    override fun getItemCount(): Int {
        return list.size
    }

    interface SuggestedSearches {
        fun hyperlinkData(hyperlink: String?)
    }

    inner class MyViewholder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var viewName: TextView = itemView.findViewById(R.id.textView)

    }

}